salto-io/salto
# SALTO-5013 - Salesforce: Report progress during deploy

Allow adapters to report progress during deploy

- [ ] ~Review of the progress message from @tomermevorach and @pgonzaleznetwork~

---

1. Allow adapters to accept a `ProgressReporter` during deploy
2. In the Salesforce adapter, Instead of waiting for the deploy to complete, check its status once in a while and report the progress

Tests:
 - Deploy and check the logs to see the expected data is printed out
 - Change the `client.polling.interval` and `client.polling.deployTimeout` configs to 1 second and confirm the deploy times out with the correct error message.

---
_Release Notes_: 
N/A

---
_User Notifications_: 
N/A
